docs/homepage/blog/offline_reinforcement_learning_algorithm_phase1/index.md

```diff
@@ -0,0 +1,315 @@
+
+\dfig{body;CRR.png}
+
+In different tasks, $f$ has different choices:
+$$
+𝑓=\mathbb{I}[A_\theta(s,a)>0]\quad \mathnormal{or}\quad f=e^{A_\theta(s,a)/\beta}
+$$We implemented discrete CRR and continuous CRR ([link](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/master/src/ReinforcementLearningZoo/src/algorithms/offline_rl/CRR.jl)). The brief function parameters are as follows: 
+
+```julia
+mutable struct CRRLearner{Aq, At, R} <: AbstractLearner
+    ### Omit other parameters
+    approximator::Aq # Actor-Critic
+    target_approximator::At # Actor-Critic
+    policy_improvement_mode::Symbol
+    ratio_upper_bound::Float32
+    beta::Float32
+    advantage_estimator::Symbol
+    m::Int
+    continuous::Bool
+end
+```
+Parameter `continuous` stands for the type of action space. `policy_improvement_mode` is the type of the weight function $f$. If `policy_improvement_mode=:binary`, we use the first $f$ function. Otherwise, we use the second $f$ function, which needs parameter `ratio_upper_bound` (Upper bound of $f$ value) and `beta`. Besides, we provide two methods to estimate advantage function, specifing `advantage_estimator=:mean/:max`. In the discrete case, we can calculate $A(s,a)$ directly. In the continuous case, we need to sample `m` Q-values to calculate advantage function.
+
+Different action spaces will also affect the implementation of the Actor-Critic. In the discrete case, the Actor outputs logits of all actions in a given state. Gaussian networks are used to model the Actor in the continuous case.
+
+Performance curve of discrete CRR algorithm in CartPole:
+
+\dfig{body;JuliaRL_CRR_CartPole.png}
+
+The continuous CRR algorithm still has some bugs and poor performance. 
+
+##### Policy in the Latent Action Space (PLAS)
+PLAS\dcite{DBLP:journals/corr/abs-2011-07213} is a policy constrain method suitable for continuous control tasks. Unlike BCQ and BEAR, PLAS implicitly constrains the policy to output actions within the support of the behavior policy through the latent action space:
+
+\dfig{body;PLAS1.png}
+
+PLAS pre-trains a CVAE (Conditional Variational Auto-Encoder) to constrain policy. In the pre-training phase, PLAS samples state-action pairs to train CVAE. PLAS needs to learn a deterministic policy mapping state to latent action and then uses CVAE mapping latent action to action in the training phase. When PLAS mapping state or latent action, it needs to use `tanh` function to limit the output range. 
+
+The advantage of pre-training VAE is that it can accelerate the convergence, and it is easier to train when encountered with complex action spaces and import existing VAE models. Its pseudocode is as follows:
+
+\dfig{body;PLAS2.png}
+
+Please refer to this link for specific code ([link](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/blob/master/src/ReinforcementLearningZoo/src/algorithms/offline_rl/PLAS.jl)). The brief function parameters are as follows:
```

Comment:
Better to use the link in docs

Reply:
What do the links in the docs? Is it a brief introduction later?

Reply:
I mean this link: https://juliareinforcementlearning.org/docs/rlzoo/#ReinforcementLearningZoo.PLASLearner-Tuple{}

Reply:
Well.